taichi/backends/opengl/codegen_opengl.cpp

```diff
@@ -467,17 +480,51 @@ class KernelGen : public IRVisitor {
       emit("{} {} = {}({});", dt_name, stmt->short_name(),
            opengl_data_type_name(stmt->cast_type), stmt->operand->short_name());
     } else if (stmt->op_type == UnaryOpType::cast_bits) {
-      if (stmt->cast_type->is_primitive(PrimitiveTypeID::f32) &&
-          stmt->operand->element_type()->is_primitive(PrimitiveTypeID::i32)) {
+      auto dst_type = stmt->cast_type;
+      auto src_type = stmt->operand->element_type();
+
+      if ((dst_type->is_primitive(PrimitiveTypeID::u32) &&
+           src_type->is_primitive(PrimitiveTypeID::i32)) ||
+          (dst_type->is_primitive(PrimitiveTypeID::u64) &&
+           src_type->is_primitive(PrimitiveTypeID::i64)) ||
+          (dst_type->is_primitive(PrimitiveTypeID::i32) &&
+           src_type->is_primitive(PrimitiveTypeID::u32)) ||
+          (dst_type->is_primitive(PrimitiveTypeID::i64) &&
+           src_type->is_primitive(PrimitiveTypeID::u64))) {
+        emit("{} {} = {}({});", dt_name, stmt->short_name(), dt_name,
+             stmt->operand->short_name());
+
+      } else if ((dst_type->is_primitive(PrimitiveTypeID::f32) &&
+                  src_type->is_primitive(PrimitiveTypeID::i32)) ||
+                 (dst_type->is_primitive(PrimitiveTypeID::f64) &&
+                  src_type->is_primitive(PrimitiveTypeID::i64))) {
         emit("{} {} = intBitsToFloat({});", dt_name, stmt->short_name(),
              stmt->operand->short_name());
-      } else if (stmt->cast_type->is_primitive(PrimitiveTypeID::i32) &&
-                 stmt->operand->element_type()->is_primitive(
-                     PrimitiveTypeID::f32)) {
+
+      } else if ((dst_type->is_primitive(PrimitiveTypeID::i32) &&
+                  src_type->is_primitive(PrimitiveTypeID::f32)) ||
+                 (dst_type->is_primitive(PrimitiveTypeID::i64) &&
+                  src_type->is_primitive(PrimitiveTypeID::f64))) {
         emit("{} {} = floatBitsToInt({});", dt_name, stmt->short_name(),
              stmt->operand->short_name());
+
+      } else if ((dst_type->is_primitive(PrimitiveTypeID::f32) &&
+                  src_type->is_primitive(PrimitiveTypeID::u32)) ||
+                 (dst_type->is_primitive(PrimitiveTypeID::f64) &&
+                  src_type->is_primitive(PrimitiveTypeID::u64))) {
+        emit("{} {} = uintBitsToFloat({});", dt_name, stmt->short_name(),
+             stmt->operand->short_name());
+
+      } else if ((dst_type->is_primitive(PrimitiveTypeID::u32) &&
+                  src_type->is_primitive(PrimitiveTypeID::f32)) ||
+                 (dst_type->is_primitive(PrimitiveTypeID::u64) &&
+                  src_type->is_primitive(PrimitiveTypeID::f64))) {
+        emit("{} {} = floatBitsToUint({});", dt_name, stmt->short_name(),
+             stmt->operand->short_name());
```

Comment:
The repeated if-condition enumerations here might not be the best way to implement a pattern like this. Given the emitted instructions all follow an "XBitsToY" structure, maybe we can simplify the code here a little bit?